src/ripple/app/ledger/impl/InboundLedger.cpp

```diff
@@ -440,14 +461,14 @@ void InboundLedger::onTimer (bool wasProgress, ScopedLockType&)
 /** Add more peers to the set, if possible */
 void InboundLedger::addPeers ()
 {
-    app_.overlay().selectPeers (*this,
+    PeerSet::addPeers(
         (getPeerCount() == 0) ? peerCountStart : peerCountAdd,
-        ScoreHasLedger (getHash(), mSeq));
+        ScoreHasLedger(mHash, mSeq));
```

Comment:
I think I wrote `ScoreHasLedger` and friends. Back then we didn't have lambdas, but now that we do, I think that we should consider using them to "inline" the code instead of requiring the user to look up a hand-rolled functor. One minor caveat: to do so _may_ require introducing functions that are currently "tucked away" in the `PeerImp` interface to the `Peer` interface. If that's the case, then it's probably not a good trade-off.